src/main/java/shef/servlets/BrowseRecipesServlet.java

```diff
@@ -14,8 +14,41 @@
 
 package shef.servlets;
 
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import shef.data.Recipe;
+import java.util.List;
+import java.util.LinkedList;
 
+@WebServlet("/browse-recipes")
 public class BrowseRecipesServlet extends HttpServlet  {
-  
-}
\ No newline at end of file
+
+  private DatastoreService datastore;
+
+  @Override
+  public void init() {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PreparedQuery recipeEntities = datastore.prepare(new Query("Recipe"));
+    List<Recipe> recipes = new LinkedList<>();
+
+    for (Entity recipeEntity : recipeEntities.asIterable()) {
+      recipes.add(new Recipe(recipeEntity));
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(recipes));
```

Comment:
When I tried to do this exact thing myself (convert a LinkedList to Json using Gson), it didn't work for me. The internet also seemed to claim that you had to use an ArrayList. I could easily be wrong though. Did you confirm that the json gets sent properly?

Reply:
I think so, when I print the json on the client side, I see exactly what I expect.